### PR TITLE
[ci] Add LUCI versions of macOS ARM tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds a macOS arm64 configuration, and adds LUCI versions of the remaining Cirrus macOS-host tests (iOS build-all and macOS platform tests) in bringup mode, to begin testing a LUCI migration for the ARM tests in this repository.

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I haven't tested with `led`, but this is the simplest of our tests (since it doesn't run anything, only builds) so I figured it was worth just giving it a try.

File: .ci.yaml [hunk 1]
```diff
@@ -74,7 +85,30 @@ targets:
       channel: stable
 
   ### iOS tasks ###
-  # TODO(stuartmorgan): Swap this and ios-build_all_plugins once simulator
+  # TODO(stuartmorgan): Swap this ios_platform_tests_* once simulator
+  # tests are reliable on the ARM infrastructure. See discussion at
+  # https://github.com/flutter/plugins/pull/5693#issuecomment-1126011089
+  - name: Mac_arm64 ios_build_all_plugins master
+    bringup: true
```
Comment (jmagman) [inline .ci.yaml:92]: If you temporarily remove `bringup: true` to force this to run in presubmit on this PR does it pass?  In the framework repo there's a check that complains if the key is missing, but I'm not sure if it'll complain in the plugins repo.

Comment (stuartmorgan) [reply .ci.yaml]: Looks like ci.yaml validation failure prevents any LUCI tests from running?

Comment (jmagman) [reply .ci.yaml]: Seems like the stable one worked: https://cirrus-ci.com/task/5239301319950336

The master one timed out fetching? https://github.com/flutter/plugins/pull/6984/checks?check_run_id=10915780299

Comment (stuartmorgan) [reply .ci.yaml]: Those are the existing Cirrus versions :)

**Comment by stuartmorgan** [comment]
The LUCI failures here are an unrelated infra problem (already being discussed on Discord).

**Comment by stuartmorgan** [comment]
I realized that since the only other macOS host test left in Cirrus besides this one is the macOS platform tests, which I expect to work if building works, I went ahead and added that to this PR too, in order to cut down on cycles of add-in-bringup-then-evaluate-then-enable.

File: .ci.yaml (hunk 1 above)
Comment (jmagman) [reply .ci.yaml]: Oh right, that's what we're doing here.

File: .ci.yaml [hunk 2]
```diff
@@ -73,17 +84,61 @@ targets:
...
+      target_file: macos_platform_tests.yaml
+
   ### iOS tasks ###
-  # TODO(stuartmorgan): Swap this and ios-build_all_plugins once simulator
-  # tests are reliable on the ARM infrastructure. See discussion at
-  # https://github.com/flutter/plugins/pull/5693#issuecomment-1126011089
+  # TODO(stuartmorgan): Swap the architecture of this and ios_platform_tests_*
+  # once simulator tests are reliable on the ARM infrastructure. See discussion
+  # at https://github.com/flutter/plugins/pull/5693#issuecomment-1126011089
+  - name: Mac_arm64 ios_build_all_plugins master
```
Comment (jmagman) [inline .ci.yaml:112]: The second space doesn't messing up the recipe script, right? I can picture a bug related to splitting on the spaces to parse `Mac_arm64`, but I didn't look.

**Review by jmagman — APPROVED** [review]

File: .ci.yaml (hunk 2 above)
Comment (stuartmorgan) [reply .ci.yaml:112]: This is the format we've been using for all of the packages and plugins LUCI tasks, and the script has been happy with it. Infra folks helped set up the initial naming structure.

**Comment by stuartmorgan** [comment]
Re-runs of these infra failures aren't triggering for some reason, but the changes to existing script files all had at least one run they passed on so I'm confident that there's no failure hidden behind the infra problem and am going to land this manually.

**Pull request merged**